addons/web/static/src/js/views/graph/graph_view.js

```diff
@@ -76,6 +80,10 @@ var GraphView = AbstractView.extend({
 
         for (const name in this.fields) {
             const field = this.fields[name];
+            const measureField = this.arch.children.find(f => f.attrs.name === name);
```

Comment:
I would prefer an implementation similar to the one you did for the pivot.
Meaning, after the loop `for (const name in measureStrings) {` here below, you would add an extra loop on `this.arch.children` and delete the keys in "measures" (and "groupableFields") that are set as invisible.
That way you only check for invisible once at the end of the process and it's consistent with the pivot code.

Reply:
Mmh that is not what I had in mind.
I meant actually create another loop statement, outside of this `for (const name in this.fields) {`, and after `for (const name in measureStrings) {`
Can you re-read my comment?

And I would prefer a "delete" from the groupBys instead of a filter, just like you did with the pivot view.